pandas/tests/frame/methods/test_drop.py

```diff
@@ -268,6 +268,30 @@ def test_drop_non_empty_list(self, index, drop_labels):
         with pytest.raises(KeyError, match="not found in axis"):
             DataFrame(index=index).drop(drop_labels)
 
+    def test_drop_empty_listlike_non_unique_datetime_index(self):
+        # GH#27994
+        data = {
+            "column_a": [5, 10, 15, 20, 25],
+            "column_b": ["one", "two", "three", "four", "five"],
+        }
+        index = [
+            Timestamp("2021-01-01"),
+            Timestamp("2021-01-01"),
+            Timestamp("2021-01-01"),
+            Timestamp("2021-01-01"),
+            Timestamp("2021-01-01"),
+        ]
+        df = DataFrame(data, index=index)
+
+        # Passing empty list or index should return the same DataFrame.
+        empty_list = []
+        empty_datetime_index = df[df["column_a"] > 100].index
```

Comment:
Please parametrize over both. Just create an empty DatetimeIndex

Reply:
I've moved the test to use pytest parameterized test functions.  We also now create our own empty DatetimeIndex instead of selecting and indexing the DataFrame to produce it.